### PR TITLE
refactor: resolve translations and entity fields together

### PR DESCRIPTION
**Pull request by briantstephan**
This adds a new `resolveComponentData` util that resolves both translations and entity fields for strings and rich text.

### CODE REVIEW

File: packages/visual-editor/src/utils/resolveTranslatableString.tsx [hunk 1]
```diff
@@ -3,9 +3,110 @@ import {
...
+ * fields, constant values, embedded fields, and translatable types.
+ *
+ * @param data The field configuration object from component props OR a direct translatable value.
+ * @param locale The current language locale (e.g., "en").
+ * @param document The entity document. If not provided, embedded fields will not be resolved.
+ * @returns The fully resolved data for the component.
+ */
+// --- Overload Signatures ---
+// 1. Handles TranslatableString directly or via a YextEntityField
+export function resolveComponentData(
```
Comment (mkilpatrick) [inline packages/visual-editor/src/utils/resolveTranslatableString.tsx:26]: This should be the only thing exported now for users to use. The old ones should be internal to us now.

File: packages/visual-editor/src/utils/resolveComponentData.test.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+          en: "Hello",
+          fr: "Bonjour",
+        },
+        constantValueEnabled: true,
+        field: "",
+      };
+      expect(resolveComponentData(field, "fr", mockDocument)).toBe("Bonjour");
+    });
+
+    // --- Embedded Field Resolution ---
```
Comment (mkilpatrick) [inline packages/visual-editor/src/utils/resolveComponentData.test.tsx:128]: Maybe you want separate describes for the main things you're testing

File: packages/visual-editor/src/utils/resolveComponentData.test.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        hasLocalizedValue: "true",
+        en: "Direct string with [[name]]",
+      };
+      expect(resolveComponentData(data, "en")).toBe(
+        "Direct string with [[name]]"
+      );
+    });
+  });
+
+  describe("edge cases", () => {
```
Comment (mkilpatrick) [inline packages/visual-editor/src/utils/resolveComponentData.test.tsx:257]: Odd name

File: packages/visual-editor/src/utils/resolveComponentData.tsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+  return newValue;
+};
+
+/**
+ * Converts a type TranslatableString to a string
+ * @param translatableString
+ * @param locale
+ * @param document
+ */
+export const resolveTranslatableString = (
```
[the remark this answers is not in the document]
Comment (mkilpatrick) [reply packages/visual-editor/src/utils/resolveComponentData.tsx:118]: We should remove unused code. Keeping this only makes it more complicated than it already is. If we ever did need it again (not sure why we would) we can check the git history.

**Review by mkilpatrick — DISMISSED** [review]

**Comment by benlife5** [comment]
```
 FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'desktop' 'version 0 props with entity values'
 FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'mobile' 'version 0 props with entity values'
 FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'desktop' 'version 7 props with entity values'
 FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'mobile' 'version 7 props with entity values'
 FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'desktop' 'version 0 props with entity values'
 FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'mobile' 'version 0 props with entity values'
 FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'desktop' 'version 7 props with entity values'
 FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'mobile' 'version 7 props with entity values'
 FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'desktop' 'version 0 props with entity values'
 FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'mobile' 'version 0 props with entity values'
 FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'desktop' 'version 7 props with entity values'
 FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'mobile' 'version 7 props with entity values'
Error: Objects are not valid as a React child (found: object with keys {html}). If you meant to render a collection of children, use an array instead.
 ❯ throwOnInvalidObjectType ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:13123:8
 ❯ createChild ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:13375:6
 ❯ reconcileChildrenArray ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:13640:24
 ❯ reconcileChildFibers2 ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:14057:15
 ❯ reconcileChildren ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:19186:27
 ❯ updateHostComponent ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:19953:2
 ❯ beginWork ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:21657:13
 ❯ beginWork$1 ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:27465:13
 ❯ performUnitOfWork ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:26599:11
 ❯ workLoopSync ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:26505:4
 ```

Do the test errors need to be addressed? Seems like it returned an object when a string or ReactNode was expected

**Comment by asanehisa** [comment]
> ```
>  FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'desktop' 'version 0 props with entity values'
>  FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'mobile' 'version 0 props with entity values'
>  FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'desktop' 'version 7 props with entity values'
>  FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'mobile' 'version 7 props with entity values'
>  FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'desktop' 'version 0 props with entity values'
>  FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'mobile' 'version 0 props with entity values'
>  FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'desktop' 'version 7 props with entity values'
>  FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'mobile' 'version 7 props with entity values'
>  FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'desktop' 'version 0 props with entity values'
>  FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'mobile' 'version 0 props with entity values'
>  FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'desktop' 'version 7 props with entity values'
>  FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'mobile' 'version 7 props with entity values'
> Error: Objects are not valid as a React child (found: object with keys {html}). If you meant to render a collection of children, use an array instead.
>  ❯ throwOnInvalidObjectType ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:13123:8
>  ❯ createChild ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:13375:6
>  ❯ reconcileChildrenArray ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:13640:24
>  ❯ reconcileChildFibers2 ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:14057:15
>  ❯ reconcileChildren ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:19186:27
>  ❯ updateHostComponent ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:19953:2
>  ❯ beginWork ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:21657:13
>  ❯ beginWork$1 ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:27465:13
>  ❯ performUnitOfWork ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:26599:11
>  ❯ workLoopSync ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:26505:4
> ```
> 
> Do the test errors need to be addressed? Seems like it returned an object when a string or ReactNode was expected

Ah agree with Ben, these were real errors.

**Comment by briantstephan** [comment]
> > ```
> >  FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'desktop' 'version 0 props with entity values'
> >  FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'mobile' 'version 0 props with entity values'
> >  FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'desktop' 'version 7 props with entity values'
> >  FAIL   chromium  src/components/pageSections/EventSection.test.tsx > EventSection > 'mobile' 'version 7 props with entity values'
> >  FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'desktop' 'version 0 props with entity values'
> >  FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'mobile' 'version 0 props with entity values'
> >  FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'desktop' 'version 7 props with entity values'
> >  FAIL   chromium  src/components/pageSections/InsightSection.test.tsx > InsightSection > 'mobile' 'version 7 props with entity values'
> >  FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'desktop' 'version 0 props with entity values'
> >  FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'mobile' 'version 0 props with entity values'
> >  FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'desktop' 'version 7 props with entity values'
> >  FAIL   chromium  src/components/pageSections/ProductSection.test.tsx > ProductSection > 'mobile' 'version 7 props with entity values'
> > Error: Objects are not valid as a React child (found: object with keys {html}). If you meant to render a collection of children, use an array instead.
> >  ❯ throwOnInvalidObjectType ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:13123:8
> >  ❯ createChild ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:13375:6
> >  ❯ reconcileChildrenArray ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:13640:24
> >  ❯ reconcileChildFibers2 ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:14057:15
> >  ❯ reconcileChildren ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:19186:27
> >  ❯ updateHostComponent ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:19953:2
> >  ❯ beginWork ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:21657:13
> >  ❯ beginWork$1 ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:27465:13
> >  ❯ performUnitOfWork ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:26599:11
> >  ❯ workLoopSync ../../node_modules/.pnpm/react-dom@18.3.1_react@18.3.1/node_modules/react-dom/cjs/react-dom.development.js:26505:4
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Do the test errors need to be addressed? Seems like it returned an object when a string or ReactNode was expected
> 
> Ah agree with Ben, these were real errors.

I see, will put up a fix.
